sycl-fusion/passes/cleanup/Cleanup.cpp

```diff
@@ -0,0 +1,164 @@
+//==---------------------------- Cleanup.cpp -------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "Cleanup.h"
+
+#include <algorithm>
+#include <utility>
+
+#include <llvm/ADT/SmallVector.h>
+#include <llvm/IR/Attributes.h>
+#include <llvm/IR/DerivedTypes.h>
+#include <llvm/IR/LLVMContext.h>
+#include <llvm/IR/Metadata.h>
+
+#include "Kernel.h"
+#include "kernel-info/SYCLKernelInfo.h"
+
+using namespace llvm;
+
+static FunctionType *createMaskedFunctionType(const BitVector &Mask,
```

Comment:
Why not anonymous namespace?

Reply:
My understanding so far was that there doesn't seem to be a big difference between `static` and anonymous namespaces, also based on [this](https://www.open-std.org/jtc1/sc22/wg21/docs/cwg_defects.html#1012). 

Do you have a specific reason to prefer one over the other?

Reply:
No strong reason, just making it more C++-y

Reply:
Using `static` in this case allows us to keep the static functions closer to their non-static users/callers without declaring multiple anonymous namespaces, so we stick with that for now.